### PR TITLE
(PUP-7471) better error message

### PR DESCRIPTION
**Pull request by DavidS**

### CODE REVIEW

File: lib/puppet/pops/parser/egrammar.ra [hunk 1]
```diff
@@ -656,6 +656,7 @@ classname
   | WORD
   | CLASSREF
   | CLASS { error val[0], "'class' is not a valid classname" }
+  | STRING { error val[0], "classnames must be bare words, not strings" }
```
Comment (hlindberg) [inline lib/puppet/pops/parser/egrammar.ra]: This is not just for the name of a class - this rule accepts `Foo`, `Foo::Bar`, `foo`, `foo::bar`, `not-name-but-word` and it is used for classes, defines, functions, etc.
Suggest changing this to be more generic "A String was found where it is not accepted"

Comment (hlindberg) [inline lib/puppet/pops/parser/egrammar.ra]: And while at it, also improve the error for the line above to be similar.

Comment (hlindberg) [inline lib/puppet/pops/parser/egrammar.ra]: simpler "A quoted string is not valid as a name here"

**Comment by DavidS** [comment]
Updated the error message as requested.

File: lib/puppet/pops/parser/egrammar.ra [hunk 2]
```diff
@@ -655,7 +655,8 @@ classname
   : NAME
   | WORD
   | CLASSREF
-  | CLASS { error val[0], "'class' is not a valid classname" }
+  | CLASS { error val[0], "'class' keyword not allowed at this location" }
+  | STRING { error val[0], "a string was found where it is not accepted" }
```
Comment (hlindberg) [inline lib/puppet/pops/parser/egrammar.ra]: you forgot to update the tests. Also, my second suggestion is better as it includes the information that the error is about a "name"

Comment (hlindberg) [inline lib/puppet/pops/parser/egrammar.ra]: my suggestion was to use "A quoted string is not acceptable as a name" - for that it is clear that it is expecting a name but got a quoted string (as opposed to a bare word which is also a string).

**Comment by DavidS** [comment]
Reading what you wrote would definitely have helped. Updated.

**Pull request merged**